### PR TITLE
Refactor RowMutationInformation to use string type

### LINKED ISSUE #31335: [Feature Request]: Enable BigQueryIO to support _CHANGE_SEQUENCE_NUMBER fixed hex string pseudo column
### What would you like to happen?

BigQuery API now provides a _CHANGE_SEQUENCE_NUMBER fixed hex string pseudo column. Request is to refactor the existing RowMutationInformation's use of long to string.

### Issue Priority

Priority: 2 (default / most feature requests should be filed as P2)

### Issue Components

- [ ] Component: Python SDK
- [X] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [X] Component: IO connector
- [ ] Component: Beam YAML
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by damondouglas**
This PR closes #31335 via refactoring BigQueryIO's assignment of the `_CHANGE_SEQUENCE_NUMBER` pseudo column from a long to fixed hex string.

All BigQuery related tests for `./gradlew :sdks:java:io:google-cloud-platform:postCommit` and `./gradlew :sdks:java:io:google-cloud-platform:build` passed on my local machine.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/AvroGenericRecordToStorageApiProto.java [hunk 1]
```diff
@@ -162,6 +162,15 @@ public static TableSchema protoTableSchemaFromAvroSchema(Schema schema) {
     return builder.build();
   }
 
+  public static DynamicMessage messageFromGenericRecord(
```
Comment (damondouglas) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/AvroGenericRecordToStorageApiProto.java:165]: Needs code comment.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiDynamicDestinationsBeamRow.java [hunk 2]
```diff
@@ -86,11 +86,11 @@ public DescriptorProtos.DescriptorProto getDescriptor(boolean includeCdcColumns)
     public StorageApiWritePayload toMessage(
         T element, @Nullable RowMutationInformation rowMutationInformation) throws Exception {
       String changeType = null;
-      long changeSequenceNum = -1;
+      String changeSequenceNum = null;
```
Comment (ahmedabu98) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiDynamicDestinationsBeamRow.java:89]: if `rowMutationInformation` is null (as it is in most use-cases), what change sequence value do we set in the DynamicMessage? Can we set a null value?

Same question for the similar changes in Avro and TableRow paths

**Review by ahmedabu98 — COMMENTED** [review]
Can we add or update existing tests in [StorageApiSinkRowUpdateIT](https://github.com/apache/beam/blob/master/sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiSinkRowUpdateIT.java)? That class contains integration tests for RowMutationInformation. Would be good to have some real end-to-end tests to check this new behavior

**Comment by damondouglas** [comment]
> Can we add or update existing tests in [StorageApiSinkRowUpdateIT](https://github.com/apache/beam/blob/master/sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiSinkRowUpdateIT.java)? That class contains integration tests for RowMutationInformation. Would be good to have some real end-to-end tests to check this new behavior

@ahmedabu98 that test already covers the refactor behavior. For example, there's a call to `RowMutationInformation.of` using a long value. This is forwarded to the `RowMutationInformation.of` that takes the string parameter set using `Long.toHexString`. Adding additional tests using the string parameter directly would just test implementation details.

**Comment by ahmedabu98** [comment]
@damondouglas Sorry I mean the new behavior of having multiple sections in the CSN (e.g. `a\b\c\d`). I don't think that's covered in our integration tests.

**Comment by Abacn** [comment]
R: @ahmedabu98

**Comment by damondouglas** [comment]
> @damondouglas Sorry I mean the new behavior of having multiple sections in the CSN (e.g. `a\b\c\d`). I don't think that's covered in our integration tests.

@ahmedabu98 I added the test cases but to keep things simple used `AAA/<n>` where `n` was the original long.

**Comment by damondouglas** [comment]
Thank you, @ahmedabu98 . I will merge after checks pass.

**Comment by damondouglas** [comment]
> Thank you, @ahmedabu98 . I will merge after checks pass.

FYI failing checks have nothing to do with code in the PR. Ran additional internal tests (see internal chat) and passed.

**Pull request merged**